test/mock_query_test.dart

```diff
@@ -1150,12 +1150,79 @@ void main() {
       'state': 'Massachusetts',
     });
 
+    await instance.collection('cities').doc().set({
+      'name': 'Washington',
+      'state': 'Washington',
+    });
+
+    final snapshots = await instance
+        .collection('cities')
+        .orderBy('name')
+        .orderBy('state')
+        .get();
+
+    expect(snapshots.docs.toData(), [
+      {
+        'name': 'Los Angeles',
+        'state': 'California',
+      },
+      {
+        'name': 'Springfield',
+        'state': 'Massachusetts',
+      },
+      {
+        'name': 'Springfield',
+        'state': 'Missouri',
+      },
+      {
+        'name': 'Springfield',
+        'state': 'Wisconsin',
+      },
+      {
+        'name': 'Washington',
+        'state': 'Washington',
+      }
+    ]);
+  });
+
+  test('startAt', () async {
+    final instance = FakeFirebaseFirestore();
+
+    await instance.collection('cities').doc().set({
+      'name': 'Los Angeles',
+      'state': 'California',
+    });
+
+    await instance.collection('cities').doc().set({
+      'name': 'Springfield',
+      'state': 'Wisconsin',
+    });
+
+    await instance.collection('cities').doc().set({
+      'name': 'Springfield',
+      'state': 'Missouri',
+    });
+
+    await instance.collection('cities').doc().set({
+      'name': 'Springfield',
+      'state': 'Massachusetts',
+    });
+
+    await instance.collection('cities').doc().set({
+      'name': 'Washington',
+      'state': 'Washington',
+    });
+
     final baseQuery =
         instance.collection('cities').orderBy('name').orderBy('state');
 
-    var snapshots = await baseQuery.startAt(['Springfield']).get();
+    var snapshots = await baseQuery.startAt(['Los Angeles']).get();
```

Comment:
What made you want to change the value here? Now we won't know if the test passes because startAt returned everything, or because it really matched with the value.

Reply:
I think I did it to be sure it returned everything when the index was 0. I don't know exactly. I will revert this.